src/Mvc/Mvc.Core/src/Resources.resx

```diff
@@ -534,4 +534,7 @@
   <data name="ValidationStrategy_MappedPropertyNotFound" xml:space="preserve">
     <value>No property found that maps to constructor parameter '{0}' for type '{1}'. Validation requires that each bound parameter of a record type's primary constructor must have a property to read the value.</value>
   </data>
-</root>
+  <data name="TryUpdateModel_RecordTypeNotSupported" xml:space="preserve">
+    <value>{0} is not supported on record type model '{1}'.</value>
```

Comment:
nit: The message sounds like only the specific `record` type is invalid at the top level when calling `TryUpdateModelAsync`. Since the restriction applies to all `record`s, I suggest
```suggestion
    <value>{0} cannot update a record type model. If a '{1}' must be updated, include it in an object type.</value>
```